### ISSUE TITLE
How do I get a en-GB translation

### ISSUE THREAD

**Issue by peterfarrant** [iLib-js/loctool#126]
We write our documentation in en-US and then need an UK English version for all the spelling differences and some technical term differences in phone system info

I can get an output of the en-GB files, but no ProjectName-new-en-GB.xliff
It appears that in PseudoFactory.js, the function 
PseudoFactory.isPseudoLocale = function (locale) 
always returns true and does not create the xliff file

But this is not for pseudo, so what am I doing wrong

[project.json.txt](https://github.com/iLib-js/loctool/files/5690722/project.json.txt)
Project file with .txt added so it is recognised for upload

**Comment by peterfarrant** [comment]
So my workaround is to just set 
```
      ispseudo = false;
      return ipseudo;
```

in line 232 of lib\PseudoFactory.js
But this is clearly not a fix
What is the intention of the PseudoFactory module ? for me it just seems to prevent creating an en-GB language xliff file

**Comment by ehoogerbeets** [comment]
Hi Peter, sorry I completely missed your question in December! Yes, right now en-GB is always a pseudo-translated locale. I will come up with a way to turn on/off pseudo-translation through the project.json or command-line.

**Comment by peterfarrant** [comment]
So my 'hack' will not cause me other issues at the moment
Our American and UK customers do not agree that it's all English :)

A way to officially allow en-GB in the project file would be good - if I could request that enhancement please

**Comment by ehoogerbeets** [comment]
For now, you can use the -n or --pseudo flag on the command-line. This turns off pseudo completely. (Unless you need pseudo for some locales, but not others.)

**Comment by ehoogerbeets** [comment]
Okay, a PR for you: https://github.com/iLib-js/loctool/pull/165

**Comment by peterfarrant** [comment]
Just been reading the updated info on locales from the file updates in the PR and that is far more capable than I was expecting/aware.

If I understand right you only do the 'auto translation translation for British-english spelling' if it is a **pseudolocale**, and in that case you create the output files but do not provide an xliff ?

It would be really good if for a 'new' element if did the 'automatic translation' for **non** pseudo locales so that the xliff already had a number of changes done before you first translate. 
If it does it only on 'new elements' in the xliff that would not cause an issue if for some reason the translation needed amending (eg British English has some fun with licence or license depending on verb or noun, but US is always license), but would save a lot of work on first pass translation - your britishSpellings.json has most of the ones I have to go and correct.

**Comment by ehoogerbeets** [comment]
Yes, that's correct. If a locale is a pseudolocale, no *-new-* xliff file will be produced. The output files are automatically generated. (Though there is still the "extracted" xliff file which only contains source strings.)

For your last paragraph there, does that mean you sort of want to do a pre-translation? That is, run the source through a British English auto-translation, and then produce the new xliff files with the source being en-US and the target pre-translated to en-GB? That might be useful if we integrate machine translation someday. Machine translation is the first pass, and then a human can just go and clean up the results rather than translating from scratch...

**Comment by peterfarrant** [comment]
**1)** 
Yes - a pre-translation would be good and save a lot of work from a translator point of view.
Seeing that you have apparently done this work if it was a pseudo locale is it possible to have it take effect for a non-pseudo version?

**2)**
Many thanks for some work on the pseudio option
I grabbed the file changes from PR165 (review pending) to try the --pseudo or -n option but not sure the results are the intended outcome from my reading of the readme
Running the command line with -n option, I thought I would get output (not translated) but I would get an xliff, but I seem to get the opposite, a pseudo version (debug translation) and no xliff for that locale
Running without the command line switch I get the pseudo output (no translation) and no xliff
I just got an error trying to use Pseudo with the type of translation eg debug, british-english... - will do some more testing later

Also the debug pseudo is taking the place holders and 'translating' those
Original Line
`* [Accounts](accounts/), Access to configure existing resellers, provision new ones`
Translated line for link
`* \<ç0>Ãççõüñţš\</ç0>, Ãççëšš ţõ çõñfíğüŕë ëxíšţíñğ ŕëšëľľëŕš, þŕõvíšíõñ ñëŵ õñëš432109876543210`

**Comment by ehoogerbeets** [comment]
It looks like the way escapes were handled was broken. This requires an update in ilib https://github.com/iLib-js/iLib/pull/292 and some more fixes in the loctool PR. We'll have to wait for the ilib changes to be approved and then we can make a new release of that.

**Comment by ehoogerbeets** [comment]
The latest version of ilib is published now (v14.9.1) which should have better escape handling. Can you try again?

**Comment by peterfarrant** [comment]
Been off for a week - sorry for the delay testing recent changes - will get onto it shortly

**Comment by peterfarrant** [comment]
Still can't get a en-GB xlif
with -n option it does seem now to have fixed the issue about localising links

**Comment by peterfarrant** [comment]
I found a workaround for now 
comment out line 251 of lib\PseudoFactory.js
```
 //   "en-GB": "english-british",
```

**Comment by peterfarrant** [comment]
Re-opened for the feature request
1)
A pre-translation would be good and save a lot of work from a translator point of view.
So if new text in a language that could have a pseudo locale translation - eg British English from US English do the pre-translation spelling corrections and then create an xliff with these in, only for new entries.
